### PR TITLE
Fix the refactored condition for pure_project

### LINKED ISSUE #9904: Factory status page doesn't show failing packages anymore
https://build.opensuse.org/projects/openSUSE:Factory/status doesn't show failing packages anymore.

Expected result:
List the failing packages like before, ie: name, devel repository, failure duration

### PR DESCRIPTION
**Pull request by saraycp**
After introducing a wrong condition on [a refactoring](https://github.com/openSUSE/open-build-service/pull/9676/commits/551b27a47b7d059d6bbcc051a39d2a55a9737a36), the failed packages were not displayed in the project status list.

Fixes #9904

### CODE REVIEW

File: src/api/spec/features/beta/webui/projects_spec.rb [hunk 1]
```diff
@@ -15,13 +15,19 @@
...
+  describe 'Project Status' do
+    before do
+      login user
+      broken_package_with_error
+      visit project_status_path(project_name: project)
+      uncheck('limit_to_fails')
+      click_button('Filter results')
+    end
+
+    it "dislays the expected packages in the results' table" do
```
Comment (danidoni) [inline src/api/spec/features/beta/webui/projects_spec.rb:27]:
```suggestion
    it "displays the expected packages in the results' table" do
```

**Review by danidoni — CHANGES_REQUESTED** [review]

File: src/api/app/models/project_status/calculator.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def calc_status(opts = {})
       @dbproj.packages.select([:id, :name, :project_id, :develpackage_id]).includes(:develpackage, :backend_package).load.each do |dbpack|
         add_recursively(mypackages, dbpack)
       end
-
+
```
Comment (danidoni) [inline src/api/app/models/project_status/calculator.rb:15]: If you remove this line the linter won't fail :D

**Review by krauselukas — COMMENTED** [review]
Besides the linter stuff and the typo that @danidoni already pointed out it looks good to me

File: src/api/spec/features/beta/webui/projects_spec.rb [hunk 3]
```diff
@@ -15,13 +15,19 @@
...
+      expect(page).to have_text('Displaying 1 packages that need handling')
```
Comment (dmarcoux) [inline src/api/spec/features/beta/webui/projects_spec.rb:28]: These 2 specs are also passing on `master`, so I'm not confident that they are really proving that the issue is fixed now with the changes included in this PR.

Comment (krauselukas) [reply src/api/spec/features/beta/webui/projects_spec.rb:28]: I think you need a package in build state 'failed' to test for the issue

Comment (krauselukas) [reply src/api/spec/features/beta/webui/projects_spec.rb:28]: without the failed comment attribute set

**Comment by saraycp** [comment]
@krauselukas @dmarcoux 

The goal of this PR is to solve a problem (P2) that is affecting our users' workflow.
As the test improvement is taking more time than expected, I'm going to push only the solution in this PR and I'll push another PR with the test soonish.

File: src/api/spec/features/beta/webui/projects_spec.rb (hunk 3 above)
Comment (saraycp) [reply src/api/spec/features/beta/webui/projects_spec.rb]: I'll take the tests out of this PR.

**Comment by krauselukas** [comment]
@saraycp fine for me, I think you need to update the cassettes in order to make rspec pass

**Review by krauselukas — APPROVED** [review]

File: src/api/spec/features/beta/webui/projects/status_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  let(:project) { user.home_project }
+  let(:broken_package_with_error) { create(:package, project: project, name: 'broken_package') }
+
+  let(:repo_for_source_project) do
+    repo = create(:repository, project: project, architectures: ['i586'])
+    project.store
+    repo
+  end
+
+  let(:local_job_history) do
```
Comment (danidoni) [inline src/api/spec/features/beta/webui/projects/status_spec.rb:14]: This is not used anywhere. Can you please remove it @saraycp ?

**Review by danidoni — CHANGES_REQUESTED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**